### ISSUE TITLE
Message system

### ISSUE THREAD

**Issue by neo22s** [open-classifieds/openclassifieds2#708]
Instead of contact form we will have a form that sends a message and goes into table of messages. This way we avoid sharing email and ad poster can choose with who to communicate. 

The button can have different texts: Apply to job offer. Counter Offer, I am interested, Contact, send a message. (make easy to change the text.)

They can add a comment and an offer. (counter offer field can be hidden).

Add new messages to the notification system (add a read field in DB).

Table structure:
id_message
id_message_parent (if null is the first one)
id_user_from
id_user_to
id_ad (not mandatory)
created
date_read
price (allow null)
message (TEXT no html no bbcode)

**Comment by neo22s** [comment]
API and model done. Frontend TODO.

**Comment by neo22s** [comment]
On yclas https://bitbucket.org/neo22s/yclas/issue/353/message-system

**Comment by neo22s** [comment]
Can we do a video or an explanation on how it works? so @ConstantinosOC can do documentation and I can test it? thanks ;)

**Comment by neo22s** [comment]
Does this use the notification system? I mean favicon + the "menu" with messages

**Comment by neo22s** [comment]
Ok, errors found:

1 - on http://reoc.lo/oc-panel/messages
![image](https://cloud.githubusercontent.com/assets/241725/8405337/037694d4-1e54-11e5-8d54-950706c16d89.png)

2 - With splash theme design issues:
http://reoc.lo/oc-panel/messages/message/17
![image](https://cloud.githubusercontent.com/assets/241725/8405364/344c4acc-1e54-11e5-8d8e-7f0a60c01330.png)

3 - If captcha is on I always get error:
ERROR El Captch no es correcto

4- If user not loged in he can fill the form and then after submitting the form I get:
INFORMACIÓN Por favor, inicia sesión antes de contactar

5- missing the messages link on the premium themes in the menu.

6 - In yclas was missing the SQL and classes to make work the messaging system I have done it now.

**Comment by oliverds** [comment]
> 1 - on http://reoc.lo/oc-panel/messages

[Fixed](https://github.com/oliverds/openclassifieds2/commit/306481ee3a130ac0342cd83d527fb26aed038ce3). Please empty your `oc2_messages` table.

> 2 - With splash theme design issues:
> http://reoc.lo/oc-panel/messages/message/17

Fixed commit on BB.

> 3 - If captcha is on I always get error:
> ERROR El Captch no es correcto

Was an issue only with splash theme. Fixed.

> 4- If user not loged in he can fill the form and then after submitting the form I get:
> INFORMACIÓN Por favor, inicia sesión antes de contactar

There was a missing commit on BB. Fixed.

> 5- missing the messages link on the premium themes in the menu.

Fixed commit on BB.

> Does this use the notification system? I mean favicon + the "menu" with messages

Yes, notification system works the same with this messages system.

> Can we do a video or an explanation on how it works? so @ConstantinosOC can do documentation and I can test it? thanks ;)

Sure.

Sorry about the errors.

**Comment by neo22s** [comment]
Awesome! no problem, thats why we do pair testing ;)

let me try again ;)

**Comment by neo22s** [comment]
When they contact them in the notification lets add a link to the message thread, no?

we have a function to get the unread messages :)

**Comment by neo22s** [comment]
I found a bug done by me commiting now

Also this
![image](https://cloud.githubusercontent.com/assets/241725/8414175/66c83618-1e96-11e5-8660-54b883fe8199.png)

Make more clear where to click to read the messages

**Comment by oliverds** [comment]
Made clickable the entire message row. It seems better.

**Comment by oliverds** [comment]
**How it works**

1. Activate messaging system http://awesomescreenshot.com/01150weu42
2. User sends message to Ad owner http://awesomescreenshot.com/08750wf111 or Users sends message to another user http://awesomescreenshot.com/08c50wg80b
3. User inbox http://awesomescreenshot.com/09050wgk4c, http://awesomescreenshot.com/0bb50wgff8
4. User can reply a message http://awesomescreenshot.com/00350wgi60

**Comment by neo22s** [comment]
@ConstantinosOC please do the documentation as soon as you can ;)

**Comment by neo22s** [comment]
@oliverds 

On yclas theme kamaleon design  broken contact form
![image](https://cloud.githubusercontent.com/assets/241725/8429003/fad73948-1f23-11e5-8ec1-54cb6efe453d.png)
